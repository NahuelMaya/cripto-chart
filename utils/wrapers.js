
import * as React from 'react'
import {render as rtlRender} from '@testing-library/react'
import {ChartContextProvider} from '../hooks/useChartContext'
import {CurrenciesContextProvider} from '../hooks/useCurrenciesContext'



function render(ui, { ...options } = {}) {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const Wrapper = ({ children }) => (
        <ChartContextProvider>
         <CurrenciesContextProvider>
            {children}
          </CurrenciesContextProvider>
        </ChartContextProvider>
    );
    return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react'

export {render}