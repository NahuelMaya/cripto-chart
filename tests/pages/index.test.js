/**
 * @jest-environment jsdom
 */
import {render, screen, waitForElementToBeRemoved} from '../../utils/wrapers'
import userEvent from '@testing-library/user-event';
import Home from "../../pages/index";
import {server} from '../../utils/tests/server'
import {rest} from 'msw'

global.ResizeObserver = require('resize-observer-polyfill')

describe("Home test", () => {
  test('Should match snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("On currencies request success, currencies should be shown", async () => {
      render(<Home />);
      expect(screen.getByText(/loading/i)).toBeInTheDocument()
      await waitForElementToBeRemoved(() => screen.getByText(/loading/i))
      expect(screen.getByText(/xrp:/i)).toBeInTheDocument()
    });

    test('Component error should show if currencies request fails', async () => {
      server.use(
        rest.get(
          'http://api.exchangeratesapi.io/v1/latest?access_key=${key}&format=1',
          async (req, res, ctx) => {
            return res(ctx.status(500))
          },
        ),
      )
      render(<Home />)
      expect(screen.getByText(/loading/i)).toBeInTheDocument()
      await waitForElementToBeRemoved(() => screen.getByText(/loading/i))
      expect(screen.getByText(/Error/i)).toBeInTheDocument()
    })
    
    test("On currency click, chart should be shown", async () => {
      render(<Home />);
      await waitForElementToBeRemoved(() => screen.getByText(/loading/i))
      userEvent.click(screen.getByText(/xrp:/i))
      expect(screen.getByText(/loading/i)).toBeInTheDocument()
      await waitForElementToBeRemoved(() => screen.getByText(/loading/i))
      expect(screen.getByRole('img')).toBeInTheDocument()
  });
});