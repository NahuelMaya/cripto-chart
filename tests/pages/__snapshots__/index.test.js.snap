// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <h1
      class="pl-5 mt-3"
    >
      CHOOSE YOUR CURRENCY
    </h1>
    <section
      class="w-full overflow-y-auto h-80 top-0 false"
    >
      <div
        class="z-50 overflow-hidden opacity-75 flex flex-col items-center justify-center "
      >
        <div
          class="loader ease-linear rounded-full border-4 border-t-4 border-gray-200 h-12 w-12 mb-4 mt-32"
        />
        <p>
          Loading
        </p>
      </div>
    </section>
    <canvas
      class="p-10 "
      height="0"
      role="img"
      style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
      width="0"
    />
  </div>
</DocumentFragment>
`;
